### ISSUE TITLE
Xaml Internal Error error WMC9999: Index (zero based) must be greater than or equal to zero and less than the size of the argument list.

### ISSUE THREAD

**Issue by omnilogix** [microsoft/WindowsAppSDK#1046]
**Describe the bug**

Getting this error when compiling a Project Reunion 0.8 Desktop app.
I was doing some refactoring, and suddenly the project won't compile.  No other obvious errors.  Any help on where to start would be appreciated.

Output:
`1>\.nuget\packages\microsoft.projectreunion.winui\0.8.0\buildTransitive\Microsoft.UI.Xaml.Markup.Compiler.interop.targets(465,9): Xaml Internal Error error WMC9999: Index (zero based) must be greater than or equal to zero and less than the size of the argument list.`

NuGet package version: 

![image](https://user-images.githubusercontent.com/9586533/125137265-335d6200-e0c1-11eb-88f1-7ec94671c07a.png)

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              |    |
| May 2020 Update (19041)            | Yes   |
| November 2019 Update (18363)       |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |

**Additional context**

**Comment by omnilogix** [comment]
So this seems to be a caching problem with Visual Studio.  After renaming the refactored pages, I was able to get normal compile errors as expected.  So, I proceeded to re-create the original pages with the original names, and the compiler gave me error messages specific to the old pages who's contents are now blank, but the error messages were for the now non-existent content, - both xaml content and code-behind content.   So, Phantom error messages!   I then deleted all of the versions of the refactored pages, cleaned the solution, re-started visual studio, and re-created the pages.  Voila!  now compiles correctly.  I have seen lots of strange behavior with error messages that don't belong to any project (no project name in output window), and don't go away after cleaning.  Only a re-start cleans them up.  This is the first time a re-start would not help.

**TLDR;**  Manual refactoring of XAML resources may break your compile cycle.  Just create new resources and add content as normal, then remove the old resources
